dm/dm/master/election_test.go

```diff
@@ -33,7 +33,6 @@ type testElectionSuite struct{}
 
 func (t *testElectionSuite) TestFailToStartLeader(c *check.C) {
 	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
```

Comment:
I think it's OK to keep this. If c.Asset in line 40 failed, there will be a temporary context leak (although no harm)